modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java

```diff
@@ -696,64 +725,137 @@ public static double defTodoubleImplicit(final Object value) {
             return (long)value;
         } else if (value instanceof Float) {
             return (float)value;
-        } else {
+        } else if (value instanceof Double) {
             return (double)value;
+        } else {
+            throw new ClassCastException("cannot implicitly " +
+                    "cast def [" + PainlessLookupUtility.typeToUnboxedType(value.getClass()).getCanonicalName() + "] to double");
         }
     }
 
     public static byte defTobyteExplicit(final Object value) {
         if (value instanceof Character) {
             return (byte)(char)value;
-        } else {
+        } else if (
+                value instanceof Byte    ||
+                value instanceof Short   ||
+                value instanceof Integer ||
+                value instanceof Long    ||
+                value instanceof Float   ||
+                value instanceof Double
+        ) {
             return ((Number)value).byteValue();
+        } else {
+            throw new ClassCastException("cannot explicitly cast " +
+                    "def [" + PainlessLookupUtility.typeToUnboxedType(value.getClass()).getCanonicalName() + "] to byte");
```

Comment:
Aren't all the boxed types handled above? when would we hit this else case? I just wonder about using typoToUnboxedType on a non-boxed type?

Reply:
Good point :).  Will fix.